pkgs/top-level/php-packages.nix

```
@@ -46,15 +46,19 @@
 , fetchpatch
 }:
 
-lib.makeScope pkgs.newScope (self: with self; {
+lib.makeScope pkgs.newScope (self: with self; let
   buildPecl = callPackage ../build-support/php/build-pecl.nix {
     php = php.unwrapped;
   };
 
-  composerHooks = callPackages ../build-support/php/hooks { };
+  builders = import ../build-support/php/builders {
+    inherit callPackages buildPecl;
+    inherit (self) callPackage;
+  };
+in {
+  inherit buildPecl;
 
-  mkComposerRepository = callPackage ../build-support/php/build-composer-repository.nix { };
-  buildComposerProject = callPackage ../build-support/php/build-composer-project.nix { };
+  inherit (builders.v1) buildComposerProject composerHooks mkComposerRepository;
```

Why not just:

```suggestion
lib.makeScope pkgs.newScope (self: with self; let
  builders = import ../build-support/php/builders {
    inherit callPackages callPackage buildPecl;
  };
in {
  buildPecl = callPackage ../build-support/php/build-pecl.nix {
    php = php.unwrapped;
  };

  inherit (builders.v1) buildComposerProject composerHooks mkComposerRepository;
```

And possibly replacing the `with self;` with `let inherit (self) /* … */;` in a preceding commit.

---

That's the part where I'm confused and where I lack of XP.

> And possibly replacing the with self; with let inherit (self) /* … */; in a preceding commit.

What am I inheriting here? What are the attributes?

---

I would start with php and add more stuff until the evaluation errors disappear.